### ISSUE TITLE
Cardano-sl build problem

### ISSUE THREAD

**Issue by mightymouse2045** [input-output-hk/cardano-sl#2139]
Hi,

I am working through the manual build steps and when i run 'nix-shell' I receive the following error:
`build input /nix/store/q8k7r5zbf03ydrpm7imfzds5b2nmdf0a-zlib-1.2.11-dev /nix/store/2xqlcc9yw8nmgiglc9w9dmcigy2mif6l-openssh-7.5p1 /nix/store/1fvizcbh34fsrj7vivan3jfmhqla4aiy-hook /nix/store/r8rj711df5n2c7pbzlcwnpfhwzvm45sw-openssl-1.0.2l-dev /nix/store/2d2in2wqdgxr0p210lv17dvxibpnfw49-gmp-6.1.2-dev /nix/store/xq5dw2qxz9b82xh53119xd40ffmwabd4-rocksdb-5.1.2 /nix/store/vzqdp5c6ki9990lywbbxyc8bnic6mcw6-git-2.14.2 /nix/store/gwzv8yf3kc751g27y602h4w9iy1anbps-bsdiff-4.3 /nix/store/zy3pprbx3226xnw9c9p58pkphfjsr8wi-happy-1.19.7 /nix/store/bj0qcghvlam4zrw9bi0h205zkvk2348m-cpphs-1.20.8 /nix/store/x7gj63y7i2451bzg5h4pnqkir5wh9q6y-cabal-install-2.0.0.0 /nix/store/3pb86cjaxmqsvar0ka3sx5qyfj2xjqxa-stack-1.5.1 /nix/store/d9ivsmlzygjzwjck8ikksh190wr3d47y-glibc-locales-2.25-49 /nix/store/rzqyfil4xmja85w5sm13m107j1zm8d7s-ghc-8.0.2 /nix/store/15axh70yqanvivgi69ns8sx7cjydqdx3-pkg-config-0.29.2 does not exist
`

I am running Fedora 27. I can't find any assistance with this anywhere - please help

**Comment by fschr** [comment]
* Are you sure /nix/store/ actually exists?
* Did you source ~/.nix-profile/etc/profile.d/nix.sh before running `nix-shell`?

**Comment by mightymouse2045** [comment]
Hi - yes I checked the paths and they all exist and I did the nix profile step as well as logging out and logging back in but same error

**Comment by mightymouse2045** [comment]
These are the instructions I am following [build cardano from source](https://github.com/input-output-hk/cardano-sl/blob/master/docs/how-to/build-cardano-sl-and-daedalus-from-source-code.md). I've installed nix and stack, building from the 1.0 branch. The DOCS there differ slightly: [1.0 DOCS](https://github.com/input-output-hk/cardano-sl/blob/cardano-sl-1.0/docs/how-to/build-cardano-sl-core-from-scratch.md), so I am going to try it that method rather than the nix-build approach first?!?

**Comment by mightymouse2045** [comment]
Nup no luck - could this be the version of nix I am running?!? Can someone let me know what version they are running to compare? Mine is: 1.11.16

**Comment by fschr** [comment]
@mightymouse2045 One trick I found to help with Nix is to just `rm -rf /nix/store`. (Obviously don't do this if you have anything else you use with Nix. If you're just trying to install Cardano-SL and Daedalus, then it's okay.)

Doing this solved lots of my problems. Specifically, certain libraries tried to link against gcc-5.4.0 instead of gcc-6.4.0 and failed. This only happened because gcc-5.4.0 came before gcc-6.4.0 in my LD_LIBRARY_PATH environment variable. I'm not sure how gcc-5.4.0 even got installed in the first place, but I deleted it from /nix/store and was able to continue the build.
